cpp/include/raft/distance/distance.cuh

```diff
@@ -415,9 +415,9 @@ void pairwise_distance(const raft::handle_t& handle,
  */
 template <typename Type, typename Index_ = int, typename layout = layout_c_contiguous>
 void pairwise_distance(raft::handle_t const& handle,
-                       device_matrix_view<Type, layout> const x,
-                       device_matrix_view<Type, layout> const y,
-                       device_matrix_view<Type, layout> dist,
+                       device_matrix_view<Type, Index_, layout> const x,
```

Comment:
If we add the `Index_` as a third type, we can default it to `std::size_t` and then we won't break any APIs downstream.

Reply:
Will do, although it will be non-breaking if default is `int` as that is what it was before